src/combo.ts

```diff
@@ -7,15 +7,45 @@ import {
   myId,
   runChoice,
 } from "kolmafia";
-import { get, set } from "libram";
+import { get, property, set } from "libram";
 
 type BeachTile = { minute: number; row: number; column: number };
 
 const rareTiles: BeachTile[] = JSON.parse(fileToBuffer("raretiles.json"));
 
 function _comb(tile: BeachTile): void {
   const { minute, row, column } = tile;
-  cliExecute(`beach wander ${minute}; beach comb ${row} ${column}`);
+  cliExecute(`beach wander ${minute};`);
+  const layout = new Map<number, string[]>(
+    property
+      .getString("_beachLayout")
+      .split(",")
+      .map((element) => element.split(":"))
+      .map((rowLayout) => [parseInt(rowLayout[0]), rowLayout[1].split("")] as [number, string[]])
+  );
+
+  const whaleRow = Array.from(layout.entries()).find((rowLayout) => rowLayout[1].includes("W"));
+  if (whaleRow) {
+    const column = whaleRow[1].find((x) => x === "W");
+    cliExecute(`beach comb ${whaleRow} ${column}`);
+    return;
+  }
+
+  const rareRow = layout.get(row);
+  if (rareRow) {
+    if (rareRow[column] !== "c") cliExecute(`beach comb ${row} ${column}`);
+  }
+
+  const firstTwinkleRow = Array.from(layout.entries()).find((rowLayout) =>
```

Comment:
I can't remember. Did we determine that twinkles that probably aren't actually a rare were worth more or less than sand?